lib/please_cli/please_cli/decision_task.py

```diff
@@ -443,16 +472,65 @@ def cmd(ctx,
         if not project_exists_in_cache:
             build_projects.append(project)
 
-    projects_to_deploy = []
+    click.echo(' => Checking github for project revisions')
+
+    project_revisions = {p: github_commit for p in PROJECTS}
+    project_url_cache = dict()
+    for project in sorted(PROJECTS):
+        # TODO: how should we handle outside projects?
+        #       skip project defined outside for now
+        if project in please_cli.config.OUTSIDE_PROJECTS:
+            continue
+
+        click.echo('     => ' + project)
+
+        project_path = '/'.join(project.split('/')[:-1])
+        project_path_end = project.split('/')[-1]
+
+        url = f'https://api.github.com/repos/mozilla/release-services/contents/src/{project_path}?ref={github_commit}'
 
+        if url in project_url_cache:
+            response = project_url_cache[url]
+        else:
+            r = requests.get(url)
+            r.raise_for_status()
+            response = r.json()
+            project_url_cache[url] = response
+
+        for item in response:
+            if project_path_end == item['name']:
+                project_revisions[project] = item['sha']
+                break
+
+    click.echo(' => Gathering deployed projects revisions')
+
+    deployed_projects = {}
+    for project in sorted(PROJECTS):
+        # TODO: how should we handle outside projects?
+        #       skip project defined outside for now
+        if project in please_cli.config.OUTSIDE_PROJECTS:
+            continue
+
+        click.echo('     => ' + project)
+
+        project = common_naming.Project(project)
+        url = f'https://index.taskcluster.net/v1/task/project.releng.services.deployment.staging.{project.taskcluster_route_name}'
+
+        r = requests.get(url)
+        r.raise_for_status()
+        response = r.json()
+
+        deployed_projects[project.name] = response['data']
+
+    projects_to_deploy = []
     if channel in please_cli.config.DEPLOY_CHANNELS:
         click.echo(' => Checking which project needs to be redeployed')
 
-        # TODO: get status for our index branch
-        deployed_projects = {}
-
         for project_name in sorted(PROJECTS):
-            deployed_projects.get(project_name)
+            if project_name in deployed_projects and \
+               project_name in project_hashes and \
+               deployed_projects[project_name]['nix_hash'] == project_hashes[project_name]:
```

Comment:
:100:

Reply:
Nit: add a `log.debug` here saying that you skipped the deploy because of a nix hash match